src/qt/test/apptests.cpp

```diff
@@ -33,13 +33,12 @@ namespace {
 //! Call getblockchaininfo RPC and check first field of JSON output.
 void TestRpcCommand(RPCConsole* console)
 {
-    QEventLoop loop;
     QTextEdit* messagesWidget = console->findChild<QTextEdit*>("messagesWidget");
-    QObject::connect(messagesWidget, &QTextEdit::textChanged, &loop, &QEventLoop::quit);
     QLineEdit* lineEdit = console->findChild<QLineEdit*>("lineEdit");
+    QSignalSpy mw_spy(messagesWidget, &QTextEdit::textChanged);
     QTest::keyClicks(lineEdit, "getblockchaininfo");
     QTest::keyClick(lineEdit, Qt::Key_Return);
-    loop.exec();
+    QVERIFY(mw_spy.wait(1000));
```

Comment:
Why 1000?

Reply:
No specific reason, just an arbitrary number. Would you suggest leaving it as the default value of 5000 milliseconds?

Reply:
I'm ok if CI passes.

Reply:
focal CI failure seems unrelated: https://cirrus-ci.com/task/5251590773800960?logs=ci#L3432